settings.py

```diff
@@ -398,25 +399,32 @@
 TERRA_WORKSPACE_CACHE_EXPIRE_SECONDS = os.environ.get('TERRA_WORKSPACE_CACHE_EXPIRE_SECONDS', 300)
 
 SERVICE_ACCOUNT_FOR_ANVIL = None
+SERVICE_ACCOUNT_CREDENTIALS = None
 
 AIRFLOW_API_AUDIENCE = os.environ.get('AIRFLOW_API_AUDIENCE')
 AIRFLOW_WEBSERVER_URL = os.environ.get('AIRFLOW_WEBSERVER_URL')
 
 if TERRA_API_ROOT_URL:
-    SERVICE_ACCOUNT_FOR_ANVIL = subprocess.run(['gcloud auth list --filter=status:ACTIVE --format="value(account)"'],
-                                               capture_output=True, text=True, shell=True).stdout.split('\n')[0] # nosec
-    if not SERVICE_ACCOUNT_FOR_ANVIL:
-        # attempt to acquire a service account token
-        if os.path.exists('/.config/service-account-key.json'):
-            auth_output = subprocess.run(['gcloud', 'auth', 'activate-service-account', '--key-file', '/.config/service-account-key.json'],  # nosec
-                                         capture_output=True, text=True).stderr
+    service_account_file = '/.config/service-account-key.json'
+    if not os.path.exists(service_account_file):
+        raise Exception('Error starting seqr - gcloud auth is not properly configured')
 
-            SERVICE_ACCOUNT_FOR_ANVIL = re.findall(r'\[(.*)\]', auth_output)[0]
+    SERVICE_ACCOUNT_CREDENTIALS = service_account.Credentials.from_service_account_file(
+        service_account_file, scopes=SOCIAL_AUTH_GOOGLE_OAUTH2_SCOPE)
+    SERVICE_ACCOUNT_FOR_ANVIL = SERVICE_ACCOUNT_CREDENTIALS.service_account_email
+    if not SERVICE_ACCOUNT_FOR_ANVIL:
+        raise Exception('Error starting seqr - gcloud auth credentials are not properly configured')
```

Comment:
Are there cases where the call to `from_service_account_file` succeeds, but we can't pull `service_account_email` off of the resulting object? If not, I feel like these lines might be cleaner in a try/except block. But, I don't feel too strongly and defer to your judgement here.

Reply:
I don't think that it is possible actually, so I will go with your suggestion